Create stabilise_re_glmer

KateEleanorx opened the pull request:
Added version of stabilise_re for logistic regression models

Other:
Can handle factors as explanatory variables
Can add a variable that is 'forced' into the model, e.g. for survival models - this will still be shown in the stability selection calculations as 'selected' or not, based on the z values > or < 2 
Option added for parallel processing of bootstraps/permutations to improve speed
Bootstrap sample process keeps the same proportion of 0/1s in the outcome variable, to prevent failure of models if no 1s are selected in the bootstrap sample

To do: seems to work with random data generated from simulate_data_re if you convert the outcome to 0/1s, I just did based on if > or < the mean outcome value

KateEleanorx committed:
Create stabilise_re_glmer

Added version of stabilise_re for logistic regression models

Other:
Can handle factors as explanatory variables
Can add a variable that is 'forced' into the model, e.g. for survival models - this will still be shown in the stability selection calculations as 'selected' or not, based on the z values > or < 2
Option for parallel processing of bootstraps/permutations
Bootstrap sample process keeps the same proportion of 0/1s in the outcome variable, to prevent failure of models if no 1s are selected in the bootstrap sample

KateEleanorx commented:
Oh, sorry also - I removed the pre-correlation bit as it was hard to do with the factors I had in the 'real' data I was using, do you think it needs adding back in?

roberthyde commented:
Hi @KateEleanorx, if you can have a look through copilot responses above, happy to merge when you're ready?

KateEleanorx committed:
LAST WORKING COMMIT

need to change the permutation function as it isn't exporting results out correctly, currently it does the same stability each time

KateEleanorx committed:
Fixed permutations error

Fixed

KateEleanorx committed:
full working glmer code with simulate data function

full working glmer code with simaulate data function

KateEleanorx committed:
Delete stabilise_re_glmer_working.R

as above

KateEleanorx committed:
Changed boot strapping to subsampling

Massively reduces the proportion of failing models when use subsampling rather than a boot-strap  with glmer
